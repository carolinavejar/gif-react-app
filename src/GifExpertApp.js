import React from 'react';
import "materialize-css/dist/css/materialize.min.css";
import { useState } from 'react/cjs/react.development';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaltCategories = []}) => {

    const [categories, setCategories] = useState(defaltCategories);

    return (
        <>
            <div className = "container">
                <h1>GifExpertApp</h1>
                <AddCategory setCategories = { setCategories }></AddCategory>
                <hr></hr>   
                    {
                        categories.map( (category) => 
                            <GifGrid 
                                category = { category }
                                key = { category }
                            >
                            </GifGrid>
                        )
                    }
            </div>
        </>
       
    );
}

export default GifExpertApp;