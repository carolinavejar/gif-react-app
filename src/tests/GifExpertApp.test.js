import React from 'react';
import { shallow } from 'enzyme';
import  GifExpertApp  from '../GifExpertApp';


describe("Pruebas de GifExpertApp", ()=> {
    test('debe de mostrar <GifExpertApp /> correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect (wrapper).toMatchSnapshot();
    });

    test('debe mostrar lista de categorias', () => {
        const categories = ['Rihanna', 'Ariana'];
        const wrapper = shallow(<GifExpertApp defaltCategories = {categories} />);
        expect (wrapper).toMatchSnapshot();
        expect ((wrapper).find('GifGrid').length).toBe(categories.length);
    });

})