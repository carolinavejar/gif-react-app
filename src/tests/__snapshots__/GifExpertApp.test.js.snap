// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifExpertApp debe de mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1>
      GifExpertApp
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
  </div>
</Fragment>
`;

exports[`Pruebas de GifExpertApp debe mostrar lista de categorias 1`] = `
<Fragment>
  <div
    className="container"
  >
    <h1>
      GifExpertApp
    </h1>
    <AddCategory
      setCategories={[Function]}
    />
    <hr />
    <GifGrid
      category="Rihanna"
      key="Rihanna"
    />
    <GifGrid
      category="Ariana"
      key="Ariana"
    />
  </div>
</Fragment>
`;
